services/powchain/src/test/java/tech/pegasys/teku/services/powchain/DepositSnapshotLoaderTest.java

```diff
@@ -0,0 +1,117 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.powchain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.assertWith;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.math.BigInteger;
+import java.net.URL;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.web3j.protocol.core.methods.response.EthBlock;
+import tech.pegasys.teku.beacon.pow.Eth1Provider;
+import tech.pegasys.teku.ethereum.pow.api.schema.LoadDepositSnapshotResult;
+import tech.pegasys.teku.ethereum.pow.api.schema.ReplayDepositsResult;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+public class DepositSnapshotLoaderTest {
+  private static final String SNAPSHOT_RESOURCE = "snapshot.ssz";
+
+  private final Spec spec = TestSpecFactory.createMinimalBellatrix();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final Eth1Provider eth1Provider = mock(Eth1Provider.class);
+
+  private String notFoundResource;
+  private DepositSnapshotLoader depositSnapshotLoader;
+
+  @BeforeEach
+  public void setup() {
+    this.notFoundResource = dataStructureUtil.randomBytes32().toHexString();
+    this.depositSnapshotLoader =
+        new DepositSnapshotLoader(
+            Optional.of(getResourceFilePath(SNAPSHOT_RESOURCE)), eth1Provider);
+  }
+
+  @Test
+  public void shouldReturnEmpty_whenNoDepositSnapshot() throws Exception {
+    this.depositSnapshotLoader = new DepositSnapshotLoader(Optional.empty(), eth1Provider);
+    final SafeFuture<LoadDepositSnapshotResult> result =
+        depositSnapshotLoader.loadDepositSnapshot();
+    assertThat(result.get().getDepositTreeSnapshot()).isEmpty();
+    verifyNoInteractions(eth1Provider);
+  }
+
+  @Test
+  public void shouldThrowInvalidConfigurationException_whenIncorrectResourcePath() {
+    this.depositSnapshotLoader =
+        new DepositSnapshotLoader(Optional.of(notFoundResource), eth1Provider);
+    final SafeFuture<LoadDepositSnapshotResult> result =
+        depositSnapshotLoader.loadDepositSnapshot();
+    assertThatThrownBy(result::get).cause().isInstanceOf(InvalidConfigurationException.class);
+  }
+
+  @Test
+  public void shouldReturnEmptyResult_whenELIsDown() throws Exception {
+    when(eth1Provider.getGuaranteedEth1Block(any(String.class)))
```

Comment:
I believe the guaranteed eth1 block method will keep retrying until it gets the block so it should never return an error (hence the `Guaranteed`).

Reply:
Interesting, funny method. The only issue I see, it will continue to cycle even if hash is incorrect and block doesn't exist. But I'd keep it as it's a best way to survive with EL starting later which doesn't look like a rare case.

Btw this one looks incorrect too https://github.com/zilm13/teku/blob/b2825238533cd0395668efcca0f51da92e90db45/beacon/pow/src/test/java/tech/pegasys/teku/beacon/pow/Eth1DepositManagerTest.java#L173-L174